File: pingyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;

@RestController
@RequestMapping("/brand")
public class BrandController {
	
	@Reference
	private BrandService brandservice;
	/**
	 * 查询所有
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbBrand> findAll() {
		
		return brandservice.findAll();
	}
	/**
	 * 分页查询
	 * @param page
	 * @param size
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult findPage(int page,int size) {
		return brandservice.findPage(page, size);
	}
	
	/**
	 * 下拉列表查询
	 * @return
	 */
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList(){
		return brandservice.selectOptionList();
	}
	/**
	 * 增加
	 * @param tbBrand
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand tbBrand) {
		try {
			brandservice.add(tbBrand);
			return new Result(true, "新加成功");
		} catch (Exception e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return new Result(false, "新加失败");
		}
		
	}
	/**
	 * 通过id查询一个
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public TbBrand findOne(long id){
		return brandservice.findOne(id);
	}
	/**
	 * 修改
	 * @param tbBrand
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand tbBrand) {
		try {
			brandservice.update(tbBrand);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
		
	}
	/**
	 * 删除
	 * @param ids
	 */
	@RequestMapping("/delete")
	public Result delete(long[] ids) {
		try {
			brandservice.delete(ids);
			return new Result(true, "删除成功");
		} catch (Exception e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
		
	}
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand tbBrand,int page,int size) {
		
		return brandservice.findPage(tbBrand, page, size);
	}
}

File: pingyougou-manager-web/src/main/webapp/js/controller/brandController.js

		app.controller('brandController',function($scope,$http,$controller,brandService){
			
			$controller('baseController',{$scope:$scope})
			//查询品牌列表
			$scope.findAll=function(){
				brandService.findAll().success(
						function(response){
							$scope.list=response;
						}
				);	
			};	
			
			
			//分页
			$scope.findPage=function(page,size){
				brandService.findPage(page,size).success(
						function(response){
							$scope.list=response.rows;//显示当前页数据
							$scope.paginationConf.totalItems=response.total;//总记录数
						}
				);	
			};	
			
			/* //增加
			$scope.save=function(){
				var methord='add';
				if($scope.entity.id!=null){
					methord='update';
				}
				$http.post('../brand/'+methord+'.do',$scope.entity).success(
						function(response){
							if(response.success){
								$scope.reloadList();//重新加载
							}else{
								alert(response.message);//错误提示
							}
						}
				);		
			};	 */
			
			//新增
			$scope.save=function(){
				var object=null;//方法名 
				if($scope.entity.id!=null){
					object=brandService.update($scope.entity);
				}else{
					object=brandService.add($scope.entity);
				}			
				object.success(
					function(response){
						if(response.success){
							$scope.reloadList();//刷新
						}else{
							alert(response.message);
						}				
					}		
				);
			}
				
			 //查询一个
			$scope.findOne=function(id){
				brandService.findOne(id).success(
					function(response){
							$scope.entity=response;
						}
					
				);
			};
			
			
			//删除
			$scope.dele=function(){
				brandService.dele($scope.selectIds).success(
						function(response){
							if(response.success){
								$scope.reloadList();//刷新
							}else{
								alert(response.message);
							}				
						}	
				);
			};
		
			
			
			//条件查询
			$scope.searchEntity={};//定义搜索对象
			$scope.search=function(page,size){
				brandService.search(page,size,$scope.searchEntity).success(
						function(response){
							$scope.list=response.rows;//显示当前页数据
							$scope.paginationConf.totalItems=response.total;//总记录数
						}
				);	
			};
//			//下拉品牌列表
//			$scope.selectOptionList=function(){
//				brandService.selectOptionList.success(
//						function(response){
//						$scope.list=response;
//				});
//			}
		});